### PR TITLE
Feat/peek

### PR DESCRIPTION
**Pull request by ThunderDev1**

### CODE REVIEW

File: app/src/components/PeekPhase.vue [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+    <div class="PeekPile">
+      <Card
+        v-for="(card, i) in cards"
+        :key="card.id"
+        :type="card.type"
+        :text="card.text"
+        class="PeekPile__Card"
+        :style="{
+          '--card-offset-x': getOffset(i).x,
+          '--card-offset-y': getOffset(i).y
```
Comment (hissalht) [inline app/src/components/PeekPhase.vue]: What I like to do in this case is to just have an "index" variable and let the CSS do its own thing with it.
```html
<Card
  v-for="(card, i) in cards"
  :style="{  --peek-phase-card-index: i }"
/>
```

```scss
// adjust these variables
$interval-x: 10rem;
$interval-y: 2rem;
$offset-x: 5rem;
$offset-y: 0;

&__Card {
  transform: 
    translateX(calc(var(--peek-phase-card-index) * #{$interval-x} + #{$offset-x}))
    translateY(calc(var(--peek-phase-card-index) * #{$interval-y} + #{$offset-y}));
}
```

File: app/src/components/Overlay.vue [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+</script>
+
+<style lang="scss">
+.Overlay {
+  position: fixed;
+  z-index: 1;
+  left: 0;
+  top: 0;
+  width: 100%;
+  height: 100%;
```
Comment (hissalht) [inline app/src/components/Overlay.vue]: Using percentage here will make the overlay have the size of the whole page. Surely, what we want is to fill the screen, not the whole page.

```suggestion
  width: 100vw;
  height: 100vh;
```

File: app/src/components/PeekPhase.vue [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+</script>
+
+<style lang="scss">
+$max-translate-offset: 0.5rem;
+.PeekPile {
+  display: flex;
+  align-items: center;
+  justify-content: center;
+  padding-top: 10%;
+  overflow: auto;
```
Comment (hissalht) [inline app/src/components/PeekPhase.vue]: No need to add padding/margins if you use `flex` on the overlay element.

```scss
.Overlay {
  display: flex;
  flex-direction: column;
  justify-content: center;
}
```

We can add this direclty in `Overlay.vue` to alway center its content or we can let components using `Overlay` decide how they are rendered.

```suggestion
```

File: app/src/components/Overlay.vue (hunk 2 above)
Comment (hissalht) [reply app/src/components/Overlay.vue]: Also, to make sure that the background cannot be scrolled while the overlay is active. We must disable scrolling on the `body` element.

```css
body {
  overflow-y: hidden;
}
```

To do this only when the overlay is rendered, we can use the `mounted` and `beforeUnmount` lifecycle hooks.

```javascript
export default {
  mounted() {
    document.body.style.overflowY = 'hidden';
  },
  beforeUnmount() {
    document.body.style.overflowY = '';
  }
}
```

**Review by hissalht — CHANGES_REQUESTED** [review]

File: app/src/components/Overlay.vue (hunk 2 above)
Comment (ThunderDev1) [reply app/src/components/Overlay.vue]: The overflow trick does not work on a mobile device. I'll use your suggestions for now, and we should manage the mobile part in another issue.

File: app/src/views/Game.vue [hunk 4]
```diff
@@ -213,7 +213,7 @@ export default {
     isPeekPhase(value, oldValue) {
       if (!value) clearInterval(this.peekCountDown);
       if (this.isUserTurn && value && !oldValue) {
-        this.peekTimeLeft = 10;
+        this.peekTimeLeft = 999;
```
Comment (hissalht) [inline app/src/views/Game.vue:216]: is this intentional ?

**Review by hissalht — APPROVED** [review]

**Pull request merged**